CompilerSource/ipc/Room.fbs

```diff
@@ -0,0 +1,113 @@
+
+table RoomInstance {
+  id:int;
+
+  x:int;
+  y:int;
+  object_id:int;
+
+  flip_x:bool;
+  flip_y:bool;
+  scale_x:double;
+  scale_y:double;
+  rotation:double;
+  color:int;
+
+  creation_code:string;
+  locked:bool;
+}
+
+struct RoomTile {
+  id:int;
+
+  background_id:int;
+  bg_x:int;
+  bg_y:int;
+  room_x:int;
+  room_y:int;
+  width:int;
+  height:int;
+  depth:int;
+
+  scale_x:double;
+  scale_y:double;
+  color:int;
+
+  locked:bool;
+}
+
+struct RoomBackground {
+  visible:bool;
+  foreground:bool;
+  background_id:int;
+  x:int;
+  y:int;
+  tile_horizontal:bool;
+  tile_vertical:bool;
+  speed_h:int;
+  speed_v:int;
+  stretch:bool;
+}
+
+struct RoomView {
+  visible:bool;
+  x:int;
+  y:int;
+  width:int;
+  height:int;
+  port_x:int;
+  port_y:int;
+  port_w:int;
+  port_h:int;
+  border_h:int;
+  border_v:int;
+  speed_h:int;
+  speed_v:int;
+  object_id:int;

```

Comment:
Another good place to reuse Point struct or the hypothetical BoundingBox struct I proposed to reuse from Sprite. Should I do this? It would make this schema a lot more concise.